code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -3147,72 +3147,75 @@
-							percent_machine +=1/2
-							if(prob(20))
-								to_chat(H, pick("<span class='warning'>Something shifts inside you...</span>", "<span class='warning'>You feel different, somehow...</span>"))
-							else
-					if(H.dizziness != 0)
-						H.dizziness = max(0, H.dizziness - 15)
-					if(H.confused != 0)
-						H.confused = max(0, H.confused - 5)
-					for(var/datum/disease/D in M.viruses)
-						D.spread = "Remissive"
-						D.stage--
-						if(D.stage < 1)
-							D.cure()
-					if(prob(percent_machine))
-						holder.add_reagent("mednanobots", 20)
-						to_chat(H, pick("<b><span class='warning'>Your body lurches!</b></span>"))
+					return
+			if(M.getOxyLoss()>0 || M.getBruteLoss()>0 || M.getToxLoss()>0 || M.getFireLoss()>0 || M.getCloneLoss()>0)
+				if(holder.has_reagent("mednanobots"))
+					M.adjustOxyLoss(-5)
+					M.heal_organ_damage(5, 5)
+					M.adjustToxLoss(-5)
+					M.adjustCloneLoss(-5)
+					holder.remove_reagent("mednanobots", 10/40)  //The number/40 means that every time it heals, it uses up number/40ths of a unit, meaning each unit heals 40 damage
+					percent_machine +=1/2
+					if(prob(20))
+						to_chat(M, pick("<span class='warning'>Something shifts inside you...</span>", "<span class='warning'>You feel different, somehow...</span>"))
+					else
+			if(M.dizziness != 0)
+				M.dizziness = max(0, M.dizziness - 15)
+			if(M.confused != 0)
+				M.confused = max(0, M.confused - 5)
+			for(var/datum/disease/D in M.viruses)
+				D.spread = "Remissive"
+				D.stage--
+				if(D.stage < 1)
+					D.cure()
+			if(prob(percent_machine))
+				holder.add_reagent("mednanobots", 20)
+				to_chat(M, pick("<b><span class='warning'>Your body lurches!</b></span>"))
 		if(20 to INFINITY)
-			if(ishuman(M))
-				spawning_horror = 1
-				var/mob/living/carbon/human/H = M
-				to_chat(H, pick("<b><span class='warning'>Something doesn't feel right...</span></b>", "<b><span class='warning'>Something is growing inside you!</span></b>", "<b><span class='warning'>You feel your insides rearrange!</span></b>"))
-				spawn(60)
-					if(spawning_horror == 1)
-						to_chat(H, "<b><span class='warning'>Something bursts out from inside you!</span></b>")
-						message_admins("[key_name(H)] has gibbed and spawned a new cyber horror due to nanobots. ([formatJumpTo(H)])")
-						var/typepath = text2path("/mob/living/simple_animal/hostile/monster/cyber_horror/[H.species.name]")
+			spawning_horror = 1
+			to_chat(M, pick("<b><span class='warning'>Something doesn't feel right...</span></b>", "<b><span class='warning'>Something is growing inside you!</span></b>", "<b><span class='warning'>You feel your insides rearrange!</span></b>"))
+			spawn(60)
+				if(spawning_horror == 1)
+					to_chat(M, "<b><span class='warning'>Something bursts out from inside you!</span></b>")
+					message_admins("[key_name(M)] has gibbed and spawned a new cyber horror due to nanobots. ([formatJumpTo(M)])")
+					if(ishuman(M))
```

Comment:
have it simply call ``new /mob/living/simple_animal/hostile/monster/cyber_horror/monster(M.loc, M)``, and do this whole "ishuman(M)" thing in cyber horror's New()